sql/src/main/java/io/crate/execution/engine/collect/stats/JobsLogService.java

```diff
@@ -89,35 +129,78 @@
+        transactionContext = TransactionContext.systemTransactionContext();
+        this.expressionAnalyzer = new ExpressionAnalyzer(
+            functions,
+            transactionContext,
+            ParamTypeHints.EMPTY,
+            new NameFieldProvider(sysJobsLogRelation),
+            null,
+            Operation.READ
+        );
+        normalizer = new EvaluatingNormalizer(functions, RowGranularity.DOC, refResolver, sysJobsLogRelation);
 
         isEnabled = STATS_ENABLED_SETTING.setting().get(settings);
         jobsLogs = new JobsLogs(this::isEnabled);
+        setUpFilter(STATS_JOBS_LOG_FILTER.setting().get(settings));
         setJobsLogSink(
-            STATS_JOBS_LOG_SIZE_SETTING.setting().get(settings), STATS_JOBS_LOG_EXPIRATION_SETTING.setting().get(settings));
+            STATS_JOBS_LOG_SIZE_SETTING.setting().get(settings),
+            STATS_JOBS_LOG_EXPIRATION_SETTING.setting().get(settings)
+        );
         setOperationsLogSink(
             STATS_OPERATIONS_LOG_SIZE_SETTING.setting().get(settings), STATS_OPERATIONS_LOG_EXPIRATION_SETTING.setting().get(settings));
 
+        clusterSettings.addSettingsUpdateConsumer(STATS_JOBS_LOG_FILTER.setting(), filter -> {
+            setUpFilter(filter);
+            updateJobSink(jobsLogSize, jobsLogExpiration);
+        });
         clusterSettings.addSettingsUpdateConsumer(STATS_ENABLED_SETTING.setting(), this::setStatsEnabled);
         clusterSettings.addSettingsUpdateConsumer(
-            STATS_JOBS_LOG_SIZE_SETTING.setting(), STATS_JOBS_LOG_EXPIRATION_SETTING.setting(), this::setJobsLogSink);
+            STATS_JOBS_LOG_SIZE_SETTING.setting(),
+            STATS_JOBS_LOG_EXPIRATION_SETTING.setting(),
+            this::setJobsLogSink);
         clusterSettings.addSettingsUpdateConsumer(
             STATS_OPERATIONS_LOG_SIZE_SETTING.setting(), STATS_OPERATIONS_LOG_EXPIRATION_SETTING.setting(), this::setOperationsLogSink);
     }
 
+    private void setUpFilter(String filterExpression) {
+        try {
+            Symbol filter = normalizer.normalize(
+                expressionAnalyzer.convert(SqlParser.createExpression(filterExpression), new ExpressionAnalysisContext()),
+                transactionContext
+            );
+            if (!filter.valueType().equals(DataTypes.BOOLEAN)) {
+                throw new IllegalArgumentException(
+                    "Filter expression for sys.jobs_log must result in a boolean, not: " + filter.valueType());
+            }
+            InputFactory.Context<RowCollectExpression<JobContextLog, ?>> ctx = inputFactory.ctxForRefs(refResolver);
+            @SuppressWarnings("unchecked")
+            Input<Boolean> filterInput = (Input<Boolean>) ctx.add(filter);
+            this.filter = tuple(filterInput, ctx.expressions());
+        } catch (Throwable t) {
+            logger.error("Could not update {}, error: {}", STATS_JOBS_LOG_FILTER.getKey(), t);
```

Comment:
from the user perspective it may be more helpful if the node won't startup when this exception occurs on config reading (surely not on `SET` statements). a logged error on node startup may be easily overseen...